impl/src/main/java/eu/darkbot/impl/managers/GalaxySpinner.java

```diff
@@ -49,13 +52,23 @@ public Optional<SpinResult> spinGate(@NotNull GalaxyGate gate, boolean multiplie
         if (multiplier) params += "&multiplier=1";
         if (spinAmount > 4) params += "&spinamount=" + spinAmount;
 
-        return Boolean.TRUE.equals(handleRequest(params, -1, minWait))
+        Optional<SpinResult> spinResult = Boolean.TRUE.equals(handleRequest(params, -1, minWait))
                 ? Optional.of(galaxyInfo.getSpinResult()) : Optional.empty();
+
+        if (!spinResult.isEmpty())
+            eventBroker.sendEvent(new SpinGateEvent(spinResult.get(), spinAmount));
+
+        return spinResult;
     }
 
     @Override
     public boolean placeGate(@NotNull GalaxyGate gate, int minWait) {
-        return Boolean.TRUE.equals(handleRequest(getParam("setupGate") + gate.getIdParam(), -1, minWait));
+        boolean successfullyPlaced = Boolean.TRUE.equals(handleRequest(getParam("setupGate") + gate.getIdParam(), -1, minWait));
+
+        if (successfullyPlaced)
+            eventBroker.sendEvent(new PlaceGateEvent(gate));
+
+        return successfullyPlaced;
```

Comment:
```suggestion
        boolean success = Boolean.TRUE.equals(handleRequest(getParam("setupGate") + gate.getIdParam(), -1, minWait));

        if (success)
            eventBroker.sendEvent(new PlaceGateEvent(gate));

        return success;
```
i'd use a sorter var name to prevent line 66 going too long